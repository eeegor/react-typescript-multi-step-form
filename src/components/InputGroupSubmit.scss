.input-submit {
	// margin: 0 auto;
	margin-top: 2rem;
}
