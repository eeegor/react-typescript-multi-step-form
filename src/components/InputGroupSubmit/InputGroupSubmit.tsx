/**
 * Component InputGroupSubmit
 */

import * as React from 'react';
import * as classnames from 'classnames';
import { FormSummary, InputGroup, Button, Label, Info } from '..';
import './InputGroupSubmit.scss';

export interface InputGroupSubmitProps {
	id: string;
	name: string;
	label?: string;
	value?: string;
	info?: string;
	type?: 'submit';
	status?: 'success' | 'danger';
	required?: boolean;
	className?: string;
	formValues?: object;
	onClick?: () => void;
}

export class InputGroupSubmit extends React.Component<InputGroupSubmitProps> {
	render(): JSX.Element {
		const {
			id,
			className,
			required,
			onClick,
			label,
			type,
			name,
			status,
			info,
			value,
			formValues,
		} = this.props;

		const publicFormFields = Object.assign({}, formValues);
		delete publicFormFields['submit'];

		return (
			<InputGroup id={id} type={'submit'} status={status} className={classnames(className)}>
				<Label className="input-group__label" htmlFor={id || name} label={label} />
				<Info className="input-group__info" label={info} />

				{publicFormFields && <FormSummary formFields={publicFormFields} />}

				<Button
					className={`input-group__input input-submit`}
					size="large"
					type="success"
					onClick={() => onClick && onClick()}
				>
					{value || 'Submit'}
				</Button>
			</InputGroup>
		);
	}
}
