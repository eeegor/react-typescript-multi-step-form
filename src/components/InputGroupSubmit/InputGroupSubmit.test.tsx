import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { InputGroupSubmit } from './InputGroupSubmit';

describe('InputGroupSubmit', () => {
	const callback = jest.fn();
	const testData = {
		id: 'submit',
		name: 'submit',
		label: 'Is this data correct?',
		value: 'Confirm your details',
		info: 'All data will be handled with care',
		onClick: callback,
	};

	it('renders markup correctly', () => {
		const component = <InputGroupSubmit {...testData} />;
		const tree = renderer.create(component).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('click event works', () => {
		const wrapper = mount(<InputGroupSubmit {...testData} />);
		wrapper.find('.button').simulate('click');
		expect(callback).toHaveBeenCalledTimes(1);
	});

	it('should adopt a valid `status: danger` attribute', () => {
		const status = 'danger';
		const wrapper = mount(<InputGroupSubmit {...testData} status={status} />);
		expect(wrapper.find('.input-group-submit').hasClass('input-group--danger')).toEqual(true);
	});

	it('should adopt a valid `status: success` attribute', () => {
		const status = 'success';
		const wrapper = mount(<InputGroupSubmit {...testData} status={status} />);
		expect(wrapper.find('.input-group-submit').hasClass('input-group--success')).toEqual(true);
	});

	it('should render default value if no value provided', () => {
		const defaultValue = 'Submit';
		const { value, ...testDataNoValue } = testData;
		const wrapper = mount(<InputGroupSubmit {...testDataNoValue} value={''} />);
		expect(wrapper.find('.button').text()).toEqual(defaultValue);
	});

	it('should render status table with correct number of elements', () => {
		const wrapper = mount(
			<InputGroupSubmit
				{...testData}
				formValues={{
					name: '',
					submit: '',
				}}
			/>
		);
		const table = wrapper.find('.form-summary .table');
		expect(table.length).toEqual(1);
		expect(table.find('.table__row__label').length).toEqual(1);
	});
});
