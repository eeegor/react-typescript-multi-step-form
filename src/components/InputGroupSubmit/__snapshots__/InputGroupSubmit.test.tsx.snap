// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputGroupSubmit renders markup correctly 1`] = `
<div
  className="input-group input-group-submit"
  id="input-group-submit"
>
  <label
    className="label input-group__label"
    htmlFor="submit"
  >
    Is this data correct?
  </label>
  <small
    className="info input-group__info"
  >
    All data will be handled with care
  </small>
  <div
    className="form-summary"
  >
    <div
      className="table"
    />
  </div>
  <button
    className="button button--success button--large input-group__input input-submit"
    onClick={[Function]}
    type="button"
  >
    Confirm your details
  </button>
</div>
`;
